src/serac/infrastructure/debug_print.hpp

```diff
@@ -121,3 +119,28 @@ void write_to_file(axom::Array<T, 3, axom::MemorySpace::Host> arr, std::string f
 
   outfile.close();
 }
+
+#ifdef __CUDACC__
+#include <cuda_runtime.h>
+/**
+ * @brief Helper function that prints usage of global device memory.  Useful for debugging potential
+ *        memory or register leaks
+ */
+#include <iostream>
+void printCUDAMemUsage()
+{
+  int deviceCount = 0;
+  cudaGetDeviceCount(&deviceCount);
+  int i = 0;
```

Comment:
We should add a TODO or have this loop over all the devices. Also it should say which MPI rank as well.